mlflow/pyfunc/__init__.py

```diff
@@ -724,6 +728,80 @@ def load_model(
     return PyFuncModel(model_meta=model_meta, model_impl=model_impl)
 
 
+def _download_model_conda_env(model_uri):
+    conda_yml_file_name = _get_flavor_configuration_from_uri(model_uri, FLAVOR_NAME)[ENV]
+    return _download_artifact_from_uri(append_to_uri_path(model_uri, conda_yml_file_name))
+
+
+def get_model_dependencies(model_uri, format="pip"):  # pylint: disable=redefined-builtin
+    """
+    Given a model URI, and format ("pip" or "conda", default value "pip"),
+    return the downloaded dependency file path.
+    If "pip" format specified but model does not have "requirements.txt" file,
+    fallback to parse the pip section of the model's "conda.yaml" and ignore other
+    non-pip dependencies in the "conda.yaml" file.
+    """
+    req_file_uri = append_to_uri_path(model_uri, _REQUIREMENTS_FILE_NAME)
+
+    if format == "pip":
+        try:
+            pip_file_path = _download_artifact_from_uri(req_file_uri)
+        except Exception as e:
+            # fallback to download conda.yaml file and parse the "pip" section from it.
+            _logger.info(
+                f"Download model '{_REQUIREMENTS_FILE_NAME}' file failed, error is {repr(e)}. "
+                "Fallback to fetch pip requirements from the model's 'conda.yaml' file and "
+                "other conda dependencies will be ignored."
+            )
+            conda_yml_path = _download_model_conda_env(model_uri)
+            pip_deps = None
+            conda_deps = []
+            try:
```

Comment:
```suggestion
        conda_yml_path = _download_model_conda_env(model_uri)
        pip_deps = None
        conda_deps = []
        try:
```

I think we can unindent this block to avoid nested try-except.